scripts/genpinctrl/config.yaml

```diff
@@ -0,0 +1,139 @@
+# Configuration file for pinctrl generation (except F1 series).
+#
+# This file contains a list of pin configuration templates used to generate the
+# pinctrl files. Each entry can have the following fields:
+#
+#   - name (mandatory): This is the pin alternate function name, e.g. UART_TX.
+#     It is used to group pin configurations alphabetically in the generated
+#     pinctrl files.
+#
+#   - match (mandatory): This is a regular expression used to match against
+#     CubeMX pin configuration names. The regular expression should be as
+#     precise as possible. Note that it needs to be escaped here in the
+#     configuration file.
+#
+#   - bias (optional): Bias setting (disable, pull-up, pull-down). If not set
+#     pinctrl file will not contain the bias entry, equivalent to disable
+#     setting.
+#
+#   - drive (optional): Drive setting (push-pull, open-drain). If not set
+#     pinctrl file will not contain the drive entry, equivalent to push-pull
+#     setting
+#
+#   - variant (optional): Define another variant of a pin configuration. This
+#     setting can be useful to define multiple configurations for a certain
+#     alternate function.
+#
+
+---
+- name: ADC_IN / ADC_INN / ADC_INP
+  match: "^ADC(?:\\d+)?_IN[NP]?\\d+$"
+
+- name: CAN_RX
+  match: "^CAN\\d+_RX$"
+  bias: pull-up
+
+- name: CAN_TX
+  match: "^CAN\\d+_TX$"
+
+- name: DAC_OUT
+  match: "^DAC(?:\\d+)?_OUT\\d+$"
+
```

Comment:
I guess the analog is selected through the script.
Would there be a way to add a mode similarly to f1 ?
Reason is that CAN_TX has no option, meaning Alternate/Push-Pull/No Pull
Here no option leads to analog...

Reply:
In case of non-F1 series a pin will be configured either as alternate or analog. Using XML database this setting can be fully determined, so a `mode` field will be redundant. F1 is special because it requires the "input" case.

Reply:
There's an interesting point: analog mode seems to be used for low-power settings, see e.g. https://wiki.st.com/stm32mpu/wiki/LPTIM_device_tree_configuration. If this is true, then a `mode` field is required.

Reply:
> Using XML database this setting can be fully determined, so a mode field will be redundant.

Sure, but this is confusing for the reader since two similar pin configurations lead to different pinctrls in the end.
Maybe a comment field (discarded at processing) would be useful for clarity.

> If this is true, then a mode field is required.

Right, we'll need this field for lp variants when we need to force pins to analog, and potentially other usages.

Reply:
Added `mode` field